### PR TITLE
Anthony/add nmea gst message

### PR DESCRIPTION
**Pull request by anth-cole**
This PR is doing 4 main things - each commit does the following 

commit 0 - Add vscode to gitignore file
commit 1 - Minor code quality changes suggested here https://swift-nav.atlassian.net/browse/FRD-94
commit 2 - Change the input from the pos_llh message to the pos_llh_cov message, mostly to support the following commit
commit 3 - Add support for the NMEA GST message detailed here https://swift-nav.atlassian.net/browse/FRD-20 - With the one caveat that we will not populate the pseudorange RMS field.

@nsirola could you in particular review the math I've implemented to construct the various GST message fields? I'm very uncertain this is correct.

### CODE REVIEW

File: c/src/nmea.c [hunk 1]
```diff
@@ -860,6 +860,78 @@ void send_gpzda(const sbp2nmea_t *state) {
...
+  
+  /* Compute the eigenvalues to get the semi-major and semi-minor axis of error ellipse */
+  double std_n = sbp_pos_llh_cov->cov_n_n * sbp_pos_llh_cov->cov_n_n;
+  double std_e = sbp_pos_llh_cov->cov_e_e * sbp_pos_llh_cov->cov_e_e;
+  double std_n_e = sbp_pos_llh_cov->cov_n_e * sbp_pos_llh_cov->cov_n_e;
+  double trace = std_n + std_e;
+  double det = std_n * std_e - std_n_e * std_n_e;
+  double semi_major = trace/2 + sqrt(((trace*trace)/4)-det);
+  double semi_minor = trace/2 - sqrt(((trace*trace)/4)-det);
+  double orientation = 90 - atan((semi_major - det)/sbp_pos_llh_cov->cov_n_e);
```
Comment (GuillaumeDec) [inline c/src/nmea.c]: - i believe the correct formula is `90 - atan((semi_major - std_e)/sbp_pos_llh_cov->cov_n_e)`
- also, atan( sbp_pos_llh_cov->cov_n_e / (semi_major - std_e) ) will give you the same withtout the extra subtraction.

**Comment by anth-cole** [comment]
ok - think this is now in a complete state

File: c/src/nmea.c [hunk 2]
```diff
@@ -855,6 +860,82 @@ void send_gpzda(const sbp2nmea_t *state) {
...
+        NMEA_SENTENCE_PRINTF("$%sGST,%s,",talker_id_to_str(i),utc);
+        break;
+      }
+    }
+  }
+  /* Currently we have no way of calculating the RMS of the observation residuals at this point so we leave it blank */
+  NMEA_SENTENCE_PRINTF(",");
+  
+  /* Compute the eigenvalues to get the semi-major and semi-minor axis of error ellipse */
+  double std_n = sbp_pos_llh_cov->cov_n_n * sbp_pos_llh_cov->cov_n_n;
```
Comment (GuillaumeDec) [inline c/src/nmea.c]:
```
  double cov_n_n = sbp_pos_llh_cov->cov_n_n;
  double cov_e_e = sbp_pos_llh_cov->cov_e_e;
  double cov_n_e = sbp_pos_llh_cov->cov_n_e;
  double std_d = sqrt(sbp_pos_llh_cov->cov_d_d);

  double trace = cov_n_n + cov_e_e;
  double det = cov_n_n * cov_e_e - cov_n_e * cov_n_e;
  // semi-major and semi-minor unchanged
  // there was an error in the orientation formula 
  double orientation = atan(cov_n_e/((trace/2 + sqrt(((trace*trace)/4)-det)) - *cov_e_e*));
```

Comment (anth-cole) [reply c/src/nmea.c]: thanks! - updated

File: c/src/nmea.c [hunk 3]
```diff
@@ -855,6 +860,86 @@ void send_gpzda(const sbp2nmea_t *state) {
...
+  }
+
+  /* Check if no SVs identified */
+  if (0 == constellations) {
+    /* At bare minimum, print empty GPGST and be done with it */
+    NMEA_SENTENCE_PRINTF("$GPGST,%s,,,,,,,",utc);
+    NMEA_SENTENCE_DONE(state);
+    return;
+  } else if (constellations > 1) {
+    /* At bare minimum, print empty GPGST and be done with it */
```
[the remark this answers is not in the document]
Comment (anth-cole) [reply c/src/nmea.c]: thanks - fixed

**Pull request merged**